hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -1141,9 +1142,11 @@ private void initializeRatisServer() throws IOException {
     if (scmRatisServer == null) {
       SCMNodeDetails scmNodeDetails = SCMNodeDetails
           .initStandAlone(configuration);
-      //TODO enable Ratis ring
-      scmRatisServer = SCMRatisServer.newSCMRatisServer(configuration, this,
-          scmNodeDetails, Collections.EMPTY_LIST);
+      //TODO enable Ratis group
+      scmRatisServer = SCMRatisServer.newSCMRatisServer(configuration
+              .getObject(SCMRatisServer.SCMRatisServerConfiguration.class),
+          this, scmNodeDetails, Collections.EMPTY_LIST,
```

Comment:
Passing half created object (this) to SCMRatisServer constructor could cause subtle bugs that are tricky to get right. We should avoid it if possible.

Reply:
I have made updates to have SCM use static method for createSCM and make constructor private. SCMRatisServer will be constructed after SCM is constructed.